### PR TITLE
CDC stream data source table name wildcard support

### PR DESCRIPTION
**Pull request by azexcy**
Related #22500.

Changes proposed in this pull request:
  - Support stream data source table name using wildcard
  - Add error code
  - Add PostgreSQL example

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/data-pipeline/cdc/client/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/client/example/opengauss/Bootstrap.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public final class Bootstrap {
      * @param args args
      */
     public static void main(final String[] args) {
-        ImportDataSourceParameter importDataSourceParam = new ImportDataSourceParameter("jdbc:opengauss:localhost:5432/cdc_db?stringtype=unspecified", "gaussdb", "Root@123");
+        ImportDataSourceParameter importDataSourceParam = new ImportDataSourceParameter("jdbc:opengauss://localhost:5432/cdc_db?stringtype=unspecified", "gaussdb", "Root@123");
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/client/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/client/example/opengauss/Bootstrap.java:35]: Could we move `Bootstrap` to `example` package?

And then merge other similar implementations to here.

File: kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/config/job/CDCJobConfiguration.java [hunk 2]
```diff
@@ -36,7 +37,7 @@ public final class CDCJobConfiguration implements PipelineJobConfiguration {
     
     private final String database;
     
-    private final List<String> schemaTableNames;
+    private final Collection<String> schemaTableNames;
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/config/job/CDCJobConfiguration.java:40]: It's better to use `List` not `Collection` in yaml related configuration class

File: kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJobId.java [hunk 3]
```diff
@@ -35,11 +35,11 @@ public final class CDCJobId extends AbstractPipelineJobId {
     
     private final String databaseName;
     
-    private final List<String> tableNames;
+    private final Collection<String> tableNames;
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJobId.java:38]: Does `tableNames` mean `schemaTableNames`?

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandler.java [hunk 4]
```diff
@@ -98,6 +117,58 @@ public CDCResponse streamData(final String requestId, final StreamDataRequestBod
...
+        }
+        return ((SchemaSupportedDatabaseType) databaseType).getDefaultSchema();
+    }
+    
+    private Collection<String> getTableNamesWithoutSchema(final ShardingSphereDatabase database, final List<SchemaTable> schemaTables) {
+        Optional<SchemaTable> allTablesOptional = schemaTables.stream().filter(each -> each.getTable().equals("*")).findFirst();
+        Set<String> allTableNames = new HashSet<>(database.getSchema(database.getName()).getAllTableNames());
+        if (allTablesOptional.isPresent()) {
+            return allTableNames;
+        }
```
Comment (sandynz) [inline proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandler.java:168]: `allTableNames` could be combined in `return`

File: proxy/backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandlerTest.java [hunk 5]
```diff
@@ -91,4 +101,61 @@ public void assertStreamDataRequestFailed() {
         CDCResponse actualResponse = handler.streamData(request.getRequestId(), request.getStreamDataRequestBody(), mock(CDCConnectionContext.class), mock(Channel.class));
         assertThat(actualResponse.getStatus(), is(Status.FAILED));
     }
+    
+    @Test
+    public void assertGetSchemaTableMapWithSchema() throws NoSuchMethodException {
+        Map<String, ShardingSphereSchema> schemas = new HashMap<>();
+        schemas.put("test", mockSchema());
+        schemas.put("public", mockSchema());
+        ShardingSphereDatabase database = new ShardingSphereDatabase("sharding_db", new PostgreSQLDatabaseType(), null, null, schemas);
```
Comment (sandynz) [inline proxy/backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandlerTest.java:110]: Could `PostgreSQLDatabaseType` be `OpenGaussDatabaseType`?

File: proxy/backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandlerTest.java [hunk 6]
```diff
@@ -91,4 +101,61 @@ public void assertStreamDataRequestFailed() {
...
+    private Map<String, String> getSchemaTableMapWithSchemaResult(final ShardingSphereDatabase database, final List<SchemaTable> schemaTables) throws NoSuchMethodException {
+        return ReflectionUtil.invokeMethod(handler.getClass().getDeclaredMethod("getSchemaTableMapWithSchema", ShardingSphereDatabase.class, List.class),
+                handler, database, schemaTables);
+    }
+    
+    @Test
+    public void assertGetSchemaTableMapWithoutSchema() throws NoSuchMethodException {
+        Map<String, ShardingSphereSchema> schemas = new HashMap<>();
+        schemas.put("sharding_db", mockSchema());
+        ShardingSphereDatabase database = new ShardingSphereDatabase("sharding_db", new PostgreSQLDatabaseType(), null, null, schemas);
```
Comment (sandynz) [inline proxy/backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandlerTest.java:146]: Could `PostgreSQLDatabaseType` be `MySQLDatabaseType`?

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJobId.java (hunk 3 above)
Comment (azexcy) [reply kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJobId.java:38]: Yes

**Review by sandynz — APPROVED** [review]

**Pull request merged**